subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -129,28 +149,65 @@ uint8_t ll_scan_enable(uint8_t enable)
 
 	scan = ull_scan_is_disabled_get(SCAN_HANDLE_1M);
 	if (!scan) {
-		return BT_HCI_ERR_CMD_DISALLOWED;
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+		is_update_1m = is_scan_update(SCAN_HANDLE_1M, duration, period,
+					      &scan, &node_rx_scan_term);
+		if (!is_update_1m)
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
+		{
+			return BT_HCI_ERR_CMD_DISALLOWED;
+		}
 	}
 
 #if defined(CONFIG_BT_CTLR_ADV_EXT) && defined(CONFIG_BT_CTLR_PHY_CODED)
 	scan_coded = ull_scan_is_disabled_get(SCAN_HANDLE_PHY_CODED);
 	if (!scan_coded) {
-		return BT_HCI_ERR_CMD_DISALLOWED;
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+		is_update_coded = is_scan_update(SCAN_HANDLE_PHY_CODED,
+						 duration, period, &scan_coded,
+						 &node_rx_scan_term);
+		if (!is_update_coded)
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
+		{
+			return BT_HCI_ERR_CMD_DISALLOWED;
+		}
 	}
 
 	own_addr_type = scan_coded->own_addr_type;
 	is_coded_phy = (scan_coded->lll.phy &
 			BT_HCI_LE_EXT_SCAN_PHY_CODED);
 #endif /* CONFIG_BT_CTLR_ADV_EXT && CONFIG_BT_CTLR_PHY_CODED */
 
-	if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT) && is_coded_phy &&
-	    ((own_addr_type & 0x1) ||
-	     (scan->own_addr_type & 0x1))) {
+	if ((is_coded_phy && (own_addr_type & 0x1)) ||
+	    (!is_coded_phy && (scan->own_addr_type & 0x1))) {
 		if (!mem_nz(ll_addr_get(1, NULL), BDADDR_SIZE)) {
 			return BT_HCI_ERR_INVALID_PARAM;
 		}
 	}
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+#if defined(CONFIG_BT_CTLR_PHY_CODED)
+	if (!is_coded_phy || (scan->lll.phy & BT_HCI_LE_EXT_SCAN_PHY_1M))
+#endif /* CONFIG_BT_CTLR_PHY_CODED */
+	{
+		err = duration_period_setup(scan, duration, period,
+					    &node_rx_scan_term);
+		if (err) {
+			return err;
+		}
+	}
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT) &&
```

Comment:
This is inside an #if defined(CONFIG_BT_CTLR_ADV_EXT) block, no need to check for IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT).